### PR TITLE
Cleanup of EDMarketConnector.py

### PR DESCRIPTION
**Pull request by Athanasius**
The main aim here is to get it passing my pre-commit hooks, so flake8 and mypy primarily.

I'm leaving TODO's behind on some things for later cleanup.

### CODE REVIEW

File: EDMarketConnector.py [hunk 1]
```diff
@@ -730,27 +775,41 @@ def crewroletext(role):
...
+
         except Exception as e:
             logger.debug('Frontier CAPI Auth:', exc_info=e)
             self.status['text'] = str(e)
+
         self.cooldown()
 
-    # Handle Status event
-    def dashboard_event(self, event):
+    def dashboard_event(self, event) -> None:
```
Comment (A-UNDERSCORE-D) [inline EDMarketConnector.py:807]: Can we assume that event is at least a `Mapping[str, Any]` here?

File: EDMarketConnector.py [hunk 2]
```diff
@@ -730,27 +775,41 @@ def crewroletext(role):
...
+        """
+        Handle DashBoardEvent tk event.
+
+        Event is sent by code in dashboard.py.
+        """
         entry = dashboard.status
         if entry:
```
Comment (A-UNDERSCORE-D) [inline EDMarketConnector.py:814]: Do we want to use a guard clause here instead?
```
if not entry:
    return

<rest of code>
```

Saves an indent level

File: EDMarketConnector.py [hunk 3]
```diff
@@ -806,30 +870,38 @@ def cooldown(self):
...
         webbrowser.open('https://github.com/EDCD/EDMarketConnector/wiki/Privacy-Policy')
 
-    def help_releases(self, event=None):
+    def help_releases(self, event=None) -> None:
+        """Open Releases page in browser."""
         webbrowser.open('https://github.com/EDCD/EDMarketConnector/releases')
 
     class HelpAbout(tk.Toplevel):
+        """The applications Help > About popup."""
+
```
Comment (A-UNDERSCORE-D) [inline EDMarketConnector.py:898]: extra newline?

File: EDMarketConnector.py [hunk 4]
```diff
@@ -901,28 +973,35 @@ def __init__(self, parent):
...
 
-        def _destroy(self):
+        def _destroy(self) -> None:
+            """Set parent window's topmost appropriately as we close."""
             self.parent.wm_attributes('-topmost', config.getint('always_ontop') and 1 or 0)
             self.destroy()
             self.__class__.showing = False
 
-    def save_raw(self):
+    def save_raw(self) -> None:  # noqa: CCR001
```
Comment (A-UNDERSCORE-D) [inline EDMarketConnector.py:986]:
```suggestion
    def save_raw(self) -> None:  # noqa: CCR001 # Not easily broken up
```

File: EDMarketConnector.py [hunk 5]
```diff
@@ -256,7 +264,7 @@ def __init__(self, master):
                                       image=self.theme_icon, cursor='fleur',
                                       anchor=tk.W, compound=tk.LEFT)
             theme_titlebar.grid(columnspan=3, padx=2, sticky=tk.NSEW)
-            self.drag_offset = None
+            self.drag_offset: Tuple[Union[None, int], Union[None, int]] = (None, None)
```
[the remark this answers is not in the document]
Comment (A-UNDERSCORE-D) [reply EDMarketConnector.py:267]: If you use this dont forget to import Optional

File: EDMarketConnector.py (hunk 1 above)
Comment (Athanasius) [reply EDMarketConnector.py:807]: That results in:

```py
EDMarketConnector.py:340: error: Argument 2 to "bind_all" of "Misc" has incompatible type "Callable[[Mapping[str, Any]], None]"; expected "Optional[Callable[[Event[Misc]], Optional[Literal['break']]]]"
```

Comment (A-UNDERSCORE-D) [reply EDMarketConnector.py:807]: Oh, thats wonderful. Nevermind then

File: EDMarketConnector.py (hunk 3 above)
Comment (Athanasius) [reply EDMarketConnector.py:899]: Newline after *class* docstring is mandatory.

Annoyingly *not* having such after function docstring is mandatory.  It keeps tripping me up :P .

Comment (A-UNDERSCORE-D) [reply EDMarketConnector.py:899]: Yeah it likes catching me too

**Review by A-UNDERSCORE-D — APPROVED** [review]

**Pull request merged**